src/components/Dashboard.tsx

```diff
@@ -94,38 +85,92 @@ const TemplateComponent: React.FC<{ template: TemplateInList }> = ({ template })
   )
 }
 
-const FilterComponent: React.FC<{
+const PanelComponent: React.FC<{
   template: TemplateDetails
-  filter: Filter
-  setFiltersCount: Function
-}> = ({ template, filter, setFiltersCount: setLoadingFilters }) => {
+  filters: Filter[]
+}> = ({ template, filters }) => {
   const templateType = template.code
-  const { loading, applicationCount } = useListApplications({
-    type: templateType,
-    perPage: 1,
-    ...filter.query,
-  })
+  const [totalMatchFilter, setTotalMatchFilter] = useState<{ [key: number]: number }>(
+    filters.reduce((totalPerFilter, filter) => ({ ...totalPerFilter, [filter.id]: 0 }), {})
+  )
+
+  const filterObjectArray = filters.reduce((arrayFilters: { [key: string]: string }[], element) => {
+    if (element.query !== null) {
+      const queryObj = element.query
+      arrayFilters.push(queryObj)
+    }
+    return arrayFilters
+  }, [])
+
+  const graphQLFilterObject = {
+    templateCode: { equalToInsensitive: templateType },
+    ...constructOrObjectFilters(filterObjectArray),
+  }
+
+  const { loading, applications } = useListApplications(
+    {}, // Passing empty as graphQLFilterObject is already constructed
+    graphQLFilterObject
+  )
 
   useEffect(() => {
-    if (!loading) setLoadingFilters((currentCount: number) => currentCount + 1)
-  }, [loading])
+    if (applications) {
+      filters.forEach(({ id, query }) => {
+        if (query !== null) {
+          const queryObj = query
+          const filteredApplications = Object.entries(applications).filter(([_, application]) => {
+            // Each filter is currently delimited to a single check!
+            const key = Object.keys(queryObj)[0]
+            const value = Object.values(queryObj)[0]
+            switch (key) {
+              case 'outcome':
+                return application.outcome === value
+              case 'status':
+                return application.status === value
+              case 'reviewerAction':
+                return application.reviewerAction === value
+              case 'assignerAction':
+                return application.assignerAction === value
+              default:
+                return false
```

Comment:
> It's not the same. I thought it could be but actually no, the thing is I need to return a boolean and just doing this isn't returning the proper result:
> 
> ```
> case 'outcome':
>     application.outcome === value
>     break
> ```
> 
> No changes done. I'm happy to try something else if you think it's worth. I actually don't. like this code so much, but haven't spent much time trying to improve this. I was thinking this would have been quicker, although very long review has been applied.

(Continuing from other PR...)

Ah I see now, and I see what you mean by "limited to a single check" -- is this likely to be a problem? Are you assuming that these pre-defined filters only have one criteria in them? I guess they do currently, but it's probably not safe to assume that they always would.

I do think this should probably not be "limited to a single check", but this is going on for a while now and it's probably fine for now. Could you make a new issue for this, something along the lines of "Dashboard filters should handle multiple filter criteria" so we don't forget about it? (Unless I'm misunderstanding it?)

Reply:
Yes! I'll make an issue now